spatialmath/box.go

```diff
@@ -255,6 +255,9 @@ func (b *box) toMesh() *Mesh {
 		m.triangles = triangles
 		b.mesh = m
 	}
+	if b.mesh.pose != NewZeroPose() {
```

Comment:
Why is this here?

Either the mesh gets filled in above with a zero pose, or something else *within spatialmath* instantiates the mesh to something specific.

We shouldn't silently edit the mesh.

Reply:
fair point, changed so that its the zero pose iff the mesh at start is nil.